Automattic/wp-calypso
# Jetpack Settings: Add module deactivation actions, reducer and selectors

Part of #9171. 

This PR takes care of the generic Redux tasks related with Jetpack Settings module deactivation. In addition, it applies some small polishing tasks. Here is a full list of what's being introduced here:

* Added actions, reducer and selectors for deactivating Jetpack settings modules.
* Added tests for the above.
* Improved the existing modules reducer by introducing `createReducer` and a couple of helpers to reduce boilerplate code.
* Added minor improvements to tests and introduced some new test cases.

To test:
* Checkout this branch
* Verify all tests pass: `npm run test-client client/state/jetpack-settings/`

/cc @oskosk @johnHackworth
